### PR TITLE
Remove ONLY_SHOW_RELEVANT_REPOS setting

### PR DESCRIPTION
**Pull request by delvh**
Every user can already disable the filter manually, so the explicit setting is absolutely useless and only complicates the logic.

Previously, there was also unexpected behavior when multiple query parameters were present.

### CODE REVIEW

**Comment by delvh** [comment]
By the way, is it breaking when I remove an unused key?
The application should still start, even if that key is present, right?
There is no action a user needs to do, except optionally removing this key from their config, right?

File: routers/web/explore/repo.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 const (
 	// tplExploreRepos explore repositories page template
 	tplExploreRepos base.TplName = "explore/repos"
+	relevantReposOnlyParam = "no_filter"
```
Comment (ghost) [inline routers/web/explore/repo.go]:
```suggestion
	relevantReposOnlyParam string       = "no_filter"
```

File: routers/web/explore/repo.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ import (
 const (
 	// tplExploreRepos explore repositories page template
 	tplExploreRepos base.TplName = "explore/repos"
```
Comment (ghost) [inline routers/web/explore/repo.go]:
```suggestion
	tplExploreRepos        base.TplName = "explore/repos"
```

File: routers/web/explore/repo.go (hunk 1 above)
Comment (ghost) [reply routers/web/explore/repo.go]: SA9004: only the first constant in this group has an explicit type

File: routers/web/explore/repo.go (hunk 2 above)
Comment (ghost) [reply routers/web/explore/repo.go]: `make fmt`

**Review by ghost — CHANGES_REQUESTED** [review]
> Repositories that are forks or that have no topic, no icon, and no description are hidden.

This is not what I expected 😂

**Comment by delvh** [comment]
What do you mean?

**Comment by delvh** [comment]
Yes, since normally if neither is present, the author doesn't seem to be interested in maintaining the repo, so it can be seen as not-explore-worthy.

**Comment by delvh** [comment]
It is not a perfect heuristic (which is likely not even possible), but it is a good enough heuristic for the general use case for for example codeberg.

**Comment by delvh** [comment]
Well,
1. this PR is not supposed to do anything else other than removing the existing useless setting
2. I don't think disabling finding forks globally is a good idea. Just think of Gitea: If you have both gogs and gitea on one instance and set up correctly, gitea will never be found on the explore page.
3. You can of course still open a PR where you request to do that to see if someone else has a different opinion on this, but as I said, I don't think that's the best idea.

**Comment by Gusted** [comment]
I'm not sure if this is a good idea; it's really **breaking**. The 'ONLY_SHOW_RELEVANT REPOS' setting was added by me to prevent this logic from being enabled on private/personal Gitea instance, where they generally don't care if a repository is relevant. Now I would navigate to `/explore` and click an extra button to actually see the majority of my repository.

**Comment by delvh** [comment]
Yes, but the fact is: You still can easily.
I don't think having to click on **one visible link less** is worth the overhead of the complicated logic as it is right now.

From your statement, you seem to use the explore page to find repos on your instance (that you use often?).
That rather means that something is wrong with our UI if you need to navigate there to find it. Do I see that right?
If that is the case, we should probably do something similar to GitHub where we show the repos where a user has contributed the most on/ where the latest contribution was on the startpage.
Would that also work for your use case, or do you simply "stalk" your users just for fun?

**Review by zeripath — APPROVED** [review]
I think this is a reasonable simplification of the logic. It's just one more click to see the full repo list.

**Review by ghost — APPROVED** [review]

**Comment by delvh** [comment]
> --- FAIL: TestLinksNoLogin (0.89s)
    testlogger.go:77: 2022/12/04 20:20:44 .../web/explore/repo.go:117:RenderRepoSearch() [E] [638d011c] SearchRepository: Count: pq: operator does not exist: json <> unknown

Does anybody have an idea why the postgres tests are failing in a likely related part, even though I touched that code only marginally?

**Comment by Gusted** [comment]
> Yes, but the fact is: You still can easily.

I hate to argue here, but if I want to continue my old experience, I have to click that button every time again.



> From your statement, you seem to use the explore page to find repos on your instance (that you use often?).

No, I have a private instance with my friends and the explore page sorts by default by recent updates, so I can follow what my friends are up to (in terms of new commits), which so far I know is the only way to achieve such thing.

**Comment by lunny** [comment]
Maybe we should display the filter options in this page whatever the default value is so that users could know and could change them.

**Comment by lunny** [comment]
CI failure is related.

**Comment by delvh** [comment]
> CI failure is related.

I know. But I don't have an idea why:

> > --- FAIL: TestLinksNoLogin (0.89s)
> > testlogger.go:77: 2022/12/04 20:20:44 .../web/explore/repo.go:117:RenderRepoSearch() [E] [638d011c] SearchRepository: Count: pq: operator does not exist: json <> unknown
> 
> Does anybody have an idea why the postgres tests are failing in a likely related part, even though I touched that code only marginally?

**Comment by HackerDaGreat57** [comment]
I too am having the `SearchRepository: Count: pq: operator does not exist: json <> unknown` issue. (Using psql 15 on arm64 ubuntu.) The Explore page and the Repositories tab in the Admin page don't work.

Everything else I tested is fine so this isn't going to corrupt your database, it's just that your Explore page and the Repository management in Admin panel will not work.

**Comment by zeripath** [comment]
> I too am having the `SearchRepository: Count: pq: operator does not exist: json <> unknown` issue. (Using psql 15, gitea 1.18.0 on arm64 ubuntu.) The Explore page and the Repositories tab in the Admin page don't work.
> 
> Everything else I tested is fine so this isn't going to corrupt your database, it's just that your Explore page and the Repository management in Admin panel will not work.
> 
> If this PR is still open then how did this get into 1.18.0? This issue was nonexistent in 1.17.4 AFAIK.

That's because it's an unrelated bug that should have had its own issue opened - however I am about to open a PR to fix it.

**Comment by lflare** [comment]
Sorry to revisit this issue again @lunny, but I share the same opinion as @Gusted. It may just be one additional link to click, but that adds up over time. Perhaps there should be an administrative default setting, and a user-override setting? I'm fairly disappointed this PR was merged and no way for any kind of overrides.

**Comment by wxiaoguang** [comment]
Actually, I changed my mind since after I eat my own dogfood.

I would say that ONLY_SHOW_RELEVANT_REPOS=false is necessary for many private/enterprise instances.

I would like to invite the option back.

**Comment by wxiaoguang** [comment]
Please help to test and review 

* #23766

**Comment by wxiaoguang** [comment]
@lflare

You can use the nighly build from https://dl.gitea.com/gitea/1.19/ or docker `gitea:1.19-dev`, everything is back.

**Pull request merged**